reascripts/ReaSpeech/source/ReaSpeechMain.lua

```diff
@@ -16,38 +16,25 @@ function ReaSpeechMain:main()
   ctx = ImGui.CreateContext(ReaSpeechUI.TITLE, ReaSpeechUI.config_flags())
   Fonts:load()
   app = ReaSpeechUI.new()
+  app:open()
+
   reaper.defer(self:loop())
 end
 
 function ReaSpeechMain:loop()
-  local visible, open = false, false
-
   return function()
     ImGui.PushFont(ctx, Fonts.main)
-    app:trap(function()
-      Theme():push(ctx)
```

Comment:
So the theme has changed to being per-window? Also, it seems like the font is part of the theme, but now that association is getting split across multiple files. Related: I'd like to make the font size changeable by the user at some point.

Reply:
more that a new toolwindow can specify its own `ImGuiTheme` to apply on top of whichever ones were applied by any parent window. this call to `Theme()` now lives in `ReaSpeechUI` in the `ToolWindow.init()` as the `theme` argument.

i did have a similar feeling about fonts, but it didn't occur to me to do anything about them at the moment. thinking on it again now, i guess there's no harm in making it another optional argument to `ToolWindow.init()`? 🤔

Reply:
Yeah, sounds good. I think moving it will also have the effect of un-theming the metrics window and anything else that is not rendered inside of ReaSpeechUI. Not sure if that matters. I think of the theme, and the main font, as sort of global concepts, but it's more flexible to have them be local to a given window, since they can always still be applied the same everywhere. With font size specifically, I feel like bumping up the font size should probably have an app-wide effect, since the main issue I see right now is tiny fonts everywhere on Linux. However, I could also see wanting to independently adjust the font size in the table view.